lib/instrumentation/index.js

```diff
@@ -358,35 +368,32 @@ Instrumentation.prototype._startHook = function () {
 
   this._agent.logger.debug('adding hooks to Node.js module loader');
 
-  this._ritmHook = new RitmHook(this._patches.keys, function (
-    exports,
-    name,
-    basedir,
-  ) {
-    const enabled = self._isModuleEnabled(name);
-    var pkg, version;
-
-    const isHandlingLambda =
-      self._lambdaHandlerInfo && self._lambdaHandlerInfo.module === name;
-
-    if (!isHandlingLambda && basedir) {
-      pkg = path.join(basedir, 'package.json');
-      try {
-        version = JSON.parse(fs.readFileSync(pkg)).version;
-      } catch (e) {
-        self._agent.logger.debug(
-          'could not shim %s module: %s',
-          name,
-          e.message,
-        );
-        return exports;
+  this._ritmHook = new RitmHook(
+    this._patches.keys,
+    // Need to enable internals so we hook into sub-modules even
+    // if the main module is hooked
+    // see: https://github.com/elastic/apm-agent-nodejs/pull/3657/files#r1344193251
+    { internals: true },
+    function (exports, name, basedir) {
+      const enabled = self._isModuleEnabled(name);
+      var version;
+
+      const isHandlingLambda =
+        self._lambdaHandlerInfo && self._lambdaHandlerInfo.module === name;
+
+      if (!isHandlingLambda && basedir) {
+        version = self._getPackageVersion(name, basedir);
```

Comment:
Note for reviewer: `_getPackageVersion` returns `undefined` if there is an exception reading the package file.

Reply:
This is already handled in https://github.com/elastic/apm-agent-nodejs/pull/3658